### ISSUE TITLE
Feature Request: rename "panos_security_policies" to "panos_security_policy"

### ISSUE THREAD

**Issue by erikpaasonen** [terraform-providers/terraform-provider-panos#34]
Since only one security policy is created per resource, this would make the naming convention consistent.

**Comment by shinmog** [comment]
I don't think this is correct..  `panos_security_policies` is a resource that manages a full set of N security policies.  A resource named `panos_security_policy` would make me think that it would only be able to manage a single security policy.

What do you think?

**Comment by shinmog** [comment]
Ah, I see.  No worries!

So you're totally right, most times in Terraform you do something as a singluar thing.  The problem with security policies is that having one or two security policies that are hanging around will affect how the firewall behaves (aka - someone puts some kind of allow any-any policy before all of your carefully crafted security policies).

To prevent this from happening, the original resource to manage security policies took a "my way or the highway" approach to security policy handling:  anything that was not explicitly in the plan file would be removed.  This solves the problem of having some random security policy in an arbitrary location affecting your total security posture.

The problem with that approach was that in a lot of situations, the team that deploys the firewall is separate from the team that wants to configure the security policies.  Actually, the app developers are spread across multiple teams, and they each have their own requirements for what should and shouldn't be allowed through the firewall.  In this scenario, these teams all manage their own Terraform state files.  So in this scenario, the assumption that there would be one place where all the security rules exist does not apply, and the design actively prevented them from being able to use the provider.

To fill this gap, this is where the security policy group resources come into play.  Using groups of security policies, each app developer team can manage their own config.  Some communication between the teams is still necessary if some kind of ordered placement is needed, but functionally speaking, the provider will allow this kind of approach now.

In either case, both are clusters of rules, tho there is nothing stopping the user from having a group of one.

**Comment by erikpaasonen** [comment]
Thank you @shinmog for explaining the big picture with the security policy groups! That is exactly our current situation with disparate teams and this explanation helps a lot in properly designing our architecture.

Re-focusing on the scope of this feature request of this particular `resource`, you're using the term `security policy` in the same context that I would instead use the term `security rule`. Intuitively I'd call a collection of `security rules` a `security policy` (note singular). I think most folks coming into the Palo Alto ecosystem from a network security background would mostly call them `rules` as well.

Consider the syntax of this resource:
``` hcl
resource "panos_security_policies" "foo" {
    rule {
        ...
    }
    rule {
        ...
    }
}
```
I don't see multiple `policy` attributes or objects inside of this `resource`. I hope this helps clarify the intent of this feature request. Does a (proposed) singular `panos_security_policy` resource name still fit within the big-picture you described, where multiple policies are aggregated using groups?

**Comment by shinmog** [comment]
Yeah, you're right...  The name of the NAT resources would have to change as well..  I'll chat with HashiCorp to see if there's a way to do this change w/o breaking the world.

**Comment by shinmog** [comment]
This is fixed in the 1.3 provider.
